pages/birthdays/index.js

```diff
@@ -0,0 +1,226 @@
+    const diffDays = Math.ceil(diffTime / (1000 * 60 * 60 * 24));
+    return diffDays;
+  };
+
+  const [friends, setFriends] = useState(
+    Object.entries(state)
+      .filter(([id, friend]) => friend.birthday)
+      .map(([id, friend]) => ({
+        id,
+        friend,
+        daysUntilBirthday: calculateDaysUntilBirthday(
+          new Date(friend.birthday)
+        ),
+      }))
+      .sort((a, b) => a.daysUntilBirthday - b.daysUntilBirthday)
+  );
+
+  const [visibleFriends, setVisibleFriends] = useState(4);
+
+  const showMoreFriends = () => {
+    setVisibleFriends((prevVisibleFriends) => prevVisibleFriends + 4);
+  };
+
+  const showLessFriends = () => {
+    setVisibleFriends((prevVisibleFriends) => prevVisibleFriends - 4);
+  };
+
+  return (
+    <>
+      <Header />
+      <StyledDiv>
+        <StyledIndex>
+          <StyledBirthdayList>
+            <StyledTitleText>Upcoming birthdays</StyledTitleText>
+
+            <StyledBirthdayBox>
+              {friends
+                .slice(0, visibleFriends)
+                .map(({ id, friend, daysUntilBirthday }) => {
+                  const birthdayDate = new Date(friend.birthday);
+                  const formattedBirthday = formatDate(birthdayDate);
+                  return (
+                    <div key={id}>
+                      <StyledSingleFriend>
+                        <StyledCleanLink
+                          href={`contacts/${id}-${friend.nickname}`}
+                        >
+                          <StyledFriendBox>
+                            <StyledProfilePicture>
+                              <Image
+                                src={friend.profileIconSource}
+                                alt={friend.nickname}
+                                width={80}
+                                height={80}
+                              />
+                            </StyledProfilePicture>
+                            <StyledFriendTextBox>
+                              <StyledNickname>{friend.nickname}</StyledNickname>
+                              <StyledBirthday>
+                                {formattedBirthday}
+                              </StyledBirthday>
+                              <StyledBirthday>
+                                {friend.nickname}`s next birthday <br />
+                                is in <b>{daysUntilBirthday}</b> days!
+                              </StyledBirthday>
+                            </StyledFriendTextBox>
+                          </StyledFriendBox>
+                        </StyledCleanLink>
+                      </StyledSingleFriend>
+                    </div>
+                  );
+                })}
+            </StyledBirthdayBox>
+            <div>
+              {visibleFriends < friends.length ? (
+                <StyledShowMoreButton>
+                  <StyledButton onClick={showMoreFriends}>
+                    Show
+                    <br />
```

Comment:
Please don't use a `<br>` element for formatting - I'm 100% sure you'll find a charming css solution for it.

Reply:
I totally agree with @JessicaLoers! Especially because I thought that there is only the word "Show" and the `br` element is used to create distance to the next element below.

However, it just came to my mind that this might be a case where the `br` element is actually useful, because it create a line break between two words where it is wanted.